book/website/communication/personal-websites/personal-websites-frameworks.md

```diff
@@ -0,0 +1,54 @@
+(cm-personal-websites-frameworks)=
+# Frameworks
+
+## What is a framework
+
+In website development, a framework is a set of pre-written code, tools, and
+libraries that help developers build and maintain websites more efficiently.
+Frameworks provide a structured foundation, offering reusable code modules,
+common design patterns, and guidelines, which help streamline the development
+process and promote best practices. They essentially "frame" the development by
+managing basic functionalities and allowing developers to focus on the unique
+aspects of their projects rather than repetitive tasks.
+
+(cm-personal-websites-frameworks-benefits)=
+## Benefits of using a framework
+
+Frameworks provide several benefits in website development:
+
+* **Code Reusability**: Common functions are built-in, which
+reduces redundancy.
+* **Efficiency**: They allow faster development by eliminating the need
+to code every component from scratch.
+* **Scalability**: Frameworks provide a solid structure that can grow with the
+website.
+* **Security**: Many frameworks come with built-in security features and
+updates, which help mitigate vulnerabilities.
+
+## Popular frameworks
+
+Below mentioned are some of the commonly used frameworks. These have been
+divided into two categories depending upon whether person wants to create a
+website using low-code or high-code options.
+
+(cm-personal-websites-frameworks-nocode)=
+### No-Code/Low-Code frameworks
+
+* [GitHub (GH) profile](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/customizing-your-profile/about-your-profile)
+  * Use GH repository [beautify my GH](<https://github.com/rzashakeri/beautify-github-profile>), to customize your GH profile.
+* [Wordpress](https://wordpress.com/support/five-step-website-setup/)
+* [Wix](https://www.wix.com/)
+* [Postcards](https://github.com/seankross/postcards)
+
+(cm-personal-websites-frameworks-procode)=
+### Pro-code/High-Code frameworks
+
+* [Jekyll](https://jekyllrb.com/)
+  * [Academic pages](https://github.com/academicpages/academicpages.github.io)
```

Comment:
Maybe this could be added to the list: https://github.com/guilyx/awesome-github-pages-portfolios

Reply:
Included the suggestion in the form of a new commit.